frozen-abi/macro/src/lib.rs

```diff
@@ -395,19 +409,26 @@ fn frozen_abi_enum_type(input: ItemEnum, expected_digest: &str) -> TokenStream {
 #[cfg(RUSTC_WITH_SPECIALIZATION)]
 #[proc_macro_attribute]
 pub fn frozen_abi(attrs: TokenStream, item: TokenStream) -> TokenStream {
-    let args = parse_macro_input!(attrs as AttributeArgs);
     let mut expected_digest: Option<String> = None;
-    for arg in args {
-        match arg {
-            NestedMeta::Meta(Meta::NameValue(nv)) if nv.path.is_ident("digest") => {
-                if let Lit::Str(lit) = nv.lit {
-                    expected_digest = Some(lit.value());
-                }
-            }
-            _ => {}
+    let attrs_parser = syn::meta::parser(|meta| {
```

Comment:
til: `syn::meta::parset` and `parse_macro_input!` combo. seems fancy.

Reply:
This is the new API in syn 2.0, that replaced old `Meta` parsing code.
https://docs.rs/syn/latest/syn/meta/fn.parser.html